src/output_updater_cuda.cu

```diff
@@ -52,6 +52,66 @@ __global__ void update_delta_z_cuda(float const *mu_a, float const *var_a,
     }
 }
 
+__global__ void update_delta_z_cuda_noise(float const *mu_a, float const *var_a,
```

Comment:
I will take a look at these function when all the changes on the CPU version is done